.circleci/config.yml

```diff
@@ -437,35 +450,54 @@ jobs:
             --form "file=@${fileName}" \
 
   instrumentation_tests_device:
-    macos:
-      xcode: "12.3.0"
-    resource_class: large
+    resource_class: xlarge
+    working_directory: ~/project
+    docker:
+      - image: circleci/android:api-28
     steps:
+      - skip-for-external-pull-requests
+      - setup-google-cloud
       - checkout
-      - setup-android-macos
       - restore-gradle-cache
-      - restore-android-build-cache-macos
-      - run-gradle-cmd:
-          desc: Build Apks for instrumentation test
-          cmd: >
-            :Corona-Warn-App:assembleDeviceDebug
-            :Corona-Warn-App:assembleDeviceDebugAndroidTest
-      - save-gradle-cache
-      - save-android-build-cache-macos
-      - run-emulator-for-api:
-          apilevel: 29
-      - run-gradle-cmd:
-          desc: Run instrumentation tests on device
-          cmd: >
-            :Corona-Warn-App:connectedDeviceDebugAndroidTest
-            --info --continue
-            -Pandroid.testInstrumentationRunnerArguments.notAnnotation=testhelpers.Screenshot
-            -Pandroid.testInstrumentationRunnerArguments.clearPackageData=true
-      - kill-all-emulators
+      - restore-android-build-cache
+      - run:
+          name: "Build app APK "
+          command: ./gradlew -PdisablePreDex :Corona-Warn-App:assembleDeviceForTestersDebug
+      - run:
+          name: "Build instrumentation test APK"
+          command: ./gradlew -PdisablePreDex :Corona-Warn-App:assembleDeviceForTestersDebugAndroidTest
+      - run:
+          name: Test with Firebase Test Lab
+          command: |
+            BUCKETDIR="`date "+%Y-%m-%d-%H:%M:%S:%3N"`-${RANDOM}"
+            echo "Using bucketdir $BUCKETDIR"
+            sudo gcloud firebase test android run \
+              --type instrumentation \
+              --app Corona-Warn-App/build/outputs/apk/deviceForTesters/debug/*.apk \
+              --test Corona-Warn-App/build/outputs/apk/androidTest/deviceForTesters/debug/*.apk \
+              --results-dir ${BUCKETDIR} \
+              --results-bucket ${GOOGLE_PROJECT_ID}-circleci-android \
+              --environment-variables clearPackageData=true \
+              --test-targets "notAnnotation testhelpers.Screenshot" \
+              --timeout 20m \
+              --device-ids flame \
+              --os-version-ids 29 \
+              --locales de_DE \
+              --orientations portrait \
+              --no-record-video
+          no_output_timeout: 30m
+      - run:
+          name: Create directory to store test results
+          command: mkdir ~/project/firebase
+      - run:
+          name: Install gsutil dependency and copy test results data
+          command: |
+            sudo pip install -U crcmod
+            sudo gsutil -m cp -R -U "`sudo gsutil ls gs://${GOOGLE_PROJECT_ID}-circleci-android/${BUCKETDIR} | tail -1`*"  ~/project/firebase | true
```

Comment:
I think we could make the path simpler.
The generated path in the zip is currently `/home/circleci/project/firebase/flame-29-de_DE-portrait/`

We could try

```yaml
          sudo gsutil -m cp -R -U "`sudo gsutil ls gs://${GOOGLE_PROJECT_ID}-circleci-android/${BUCKETDIR} | tail -1`*" firebase-results | true

(...)
      - store_test_results:
          path: ./firebase-results/flame-29-de_DE-portrait
      - compress-path:
          input: ./firebase-results/flame-29-de_DE-portrait/*.xml
(...)
```

What do you think?